test/xref2/deep_substitution.t/run.t

```diff
@@ -0,0 +1,46 @@
+Test to check that when doing a deep substitution inside a signature,
+that items that reference that element are still correctly resolved
+even if they're in a different path.
+
+  $ cat m.mli
+  module type S = sig
+    module M: sig
+      type t
+    end
+    type t = M.t
+  end
+  module type T = S with type M.t := int
+  
+
+In this, we want to check that the type `t` in module type `T` has
+its RHS correctly replaced with an `int`
+
+  $ ocamlc -c -bin-annot m.mli
+  $ odoc compile m.cmti
+  $ odoc link m.odoc
+  $ odoc html-generate m.odocl --indent -o .
```

Comment:
We have a helper script for this:
```suggestion
  $ compile m.mli
```
The output is slightly different because it sets the parent. If that's annoying, perhaps we can change the script and update every other tests ?

Reply:
We do have a helper script, true - but this way the test is a bit more self contained, and I often find this useful when I'm working on them because I find building them out of tree to be essential when developing (and debugging) these tests.